src/Compilers/Core/Portable/MetadataReader/MetadataDecoder.cs

```diff
@@ -319,6 +319,18 @@ private TypeSymbol DecodeTypeOrThrow(ref BlobReader ppSig, SignatureTypeCode typ
                     typeSymbol = DecodeGenericTypeInstanceOrThrow(ref ppSig, out refersToNoPiaLocalType);
                     break;
 
+                case SignatureTypeCode.FunctionPointer:
+                    var signatureHeader = ppSig.ReadSignatureHeader();
+                    var parameters = DecodeSignatureParametersOrThrow(ref ppSig, signatureHeader, typeParameterCount: out int typeParamCount, shouldProcessAllBytes: false);
+
+                    if (typeParamCount != 0)
+                    {
+                        throw new UnsupportedSignatureContent();
+                    }
+
+                    typeSymbol = MakeFunctionPointerTypeSymbol(Cci.CallingConventionUtils.FromSignatureConvention(signatureHeader.CallingConvention), ImmutableArray.Create(parameters));
```

Comment:
>Cci.CallingConventionUtils.FromSignatureConvention(signatureHeader.CallingConvention) [](start = 63, length = 85)

`signatureHeader.CallingConvention.FromSignatureConvention()`

Reply:
Actually, there aren't that many here, I'll just do it.

---
In reply to: [375578610](https://github.com/dotnet/roslyn/pull/41340#discussion_r375578610) [](ancestors = 375578610,375063146)

Reply:
> there aren't that many here

But there _are_ conflicting names between Cci. and SRM., so I'll leave it.

---
In reply to: [375582556](https://github.com/dotnet/roslyn/pull/41340#discussion_r375582556) [](ancestors = 375582556,375578610,375063146)